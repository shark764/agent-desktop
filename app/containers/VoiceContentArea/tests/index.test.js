/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import { VoiceContentArea } from '../index';

describe('<VoiceContentArea />', () => {
  const mockInteraction = {
    interactionId: 'mock-interaction-1',
    status: 'work-accepted',
    channelType: 'work-item',
    subject: 'Test subject',
    customFields: [{ label: 'label', value: 'value' }],
    script: { something: 'something' },
    contact: { attributes: { name: 'Contact name' } },
  };

  it('should render correctly', () => {
    const rendered = shallow(
      <VoiceContentArea
        selectedInteraction={mockInteraction}
        endInteraction={() => {}}
        awaitingScript={false}
        isEndWrapupDisabled={false}
        wrapupBtnTooltipText={{}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });

  it('should render correctly when waiting on script', () => {
    const rendered = shallow(
      <VoiceContentArea
        selectedInteraction={mockInteraction}
        endInteraction={() => {}}
        awaitingScript
        isEndWrapupDisabled
        wrapupBtnTooltipText={{ id: 'testId', defaultMessage: 'test message' }}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
