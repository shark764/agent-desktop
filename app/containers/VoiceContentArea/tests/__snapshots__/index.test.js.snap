// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VoiceContentArea /> should render correctly 1`] = `
<Connect(ErrorBoundaryComponent)
  buttonConfig={
    Array [
      Object {
        "disabled": false,
        "id": "hang-up-button",
        "onClick": [Function],
        "text": Object {
          "defaultMessage": "Hang Up",
          "id": "app.containers.VoiceContentArea.hangUp",
        },
        "tooltipText": Object {},
        "type": "primaryRed",
      },
    ]
  }
  content={
    <div
      style={
        Array [
          Object {
            "height": "100%",
            "overflowY": "auto",
            "padding": "17px",
            "position": "absolute",
            "width": "100%",
          },
          false,
        ]
      }
    >
      <Connect(ErrorBoundaryComponent) />
    </div>
  }
  details={<Connect(ErrorBoundaryComponent) />}
  from="Contact name"
  interaction={
    Object {
      "channelType": "work-item",
      "contact": Object {
        "attributes": Object {
          "name": "Contact name",
        },
      },
      "customFields": Array [
        Object {
          "label": "label",
          "value": "value",
        },
      ],
      "interactionId": "mock-interaction-1",
      "script": Object {
        "something": "something",
      },
      "status": "work-accepted",
      "subject": "Test subject",
    }
  }
/>
`;

exports[`<VoiceContentArea /> should render correctly when waiting on script 1`] = `
<Connect(ErrorBoundaryComponent)
  buttonConfig={
    Array [
      Object {
        "disabled": true,
        "id": "hang-up-button",
        "onClick": [Function],
        "text": Object {
          "defaultMessage": "Hang Up",
          "id": "app.containers.VoiceContentArea.hangUp",
        },
        "tooltipText": Object {
          "defaultMessage": "test message",
          "id": "testId",
        },
        "type": "primaryRed",
      },
    ]
  }
  content={
    <div
      style={
        Array [
          Object {
            "height": "100%",
            "overflowY": "auto",
            "padding": "17px",
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "border": "1px solid #FE4565",
          },
        ]
      }
    >
      <Connect(ErrorBoundaryComponent) />
    </div>
  }
  details={<Connect(ErrorBoundaryComponent) />}
  from="Contact name"
  interaction={
    Object {
      "channelType": "work-item",
      "contact": Object {
        "attributes": Object {
          "name": "Contact name",
        },
      },
      "customFields": Array [
        Object {
          "label": "label",
          "value": "value",
        },
      ],
      "interactionId": "mock-interaction-1",
      "script": Object {
        "something": "something",
      },
      "status": "work-accepted",
      "subject": "Test subject",
    }
  }
/>
`;
